File: listing-service/src/main/java/com/p2p/listing/entity/Listing.java
package com.p2p.listing.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "listings")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Listing {
    @Id
    private UUID id;

    @Column(nullable = false)
    private UUID vehicleId;

    @Column(nullable = false)
    private UUID ownerId;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal pricePerDay;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal depositAmount;

    @Column(nullable = false)
    private boolean instantBooking;

    @Column(nullable = false)
    private boolean deliveryAvailable;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ListingStatus status;

    @Column(nullable = false)
    private Instant createdAt;

    @Column(nullable = false)
    private Instant updatedAt;

    @PrePersist
    void prePersist() {
        if (id == null) id = UUID.randomUUID();
        if (status == null) status = ListingStatus.ACTIVE;
        createdAt = Instant.now();
        updatedAt = createdAt;
    }

    @PreUpdate
    void preUpdate() {
        updatedAt = Instant.now();
    }
}

File: frontend/src/types/listing.ts
export type ListingPhoto = {
  id: string;
  photoUrl: string;
  sortOrder: number;
};

export type ListingRules = {
  minDriverAge: number;
  minRentalDays: number;
  maxRentalDays: number;
  smokingAllowed: boolean;
  petsAllowed: boolean;
  internationalTravelAllowed: boolean;
};

export type Listing = {
  id: string;
  vehicleId: string;
  ownerId: string;
  title: string;
  description: string;
  city: string;
  pricePerDay: number;
  depositAmount: number;
  instantBooking: boolean;
  deliveryAvailable: boolean;
  photos: ListingPhoto[];
  rules?: ListingRules;
};

export type ListingSummary = {
  id: string;
  vehicleId: string;
  title: string;
  city: string;
  pricePerDay: number;
  deliveryAvailable: boolean;
};
